MultiplayerTestRunner/src/Utils.cpp

```diff
@@ -0,0 +1,89 @@
+#include "Utils.h"
+
+#include "../include/ErrorCodes.h"
+
+#include <CSP/CSPFoundation.h>
+#include <CSP/Common/String.h>
+#include <CSP/Systems/WebService.h> //For resultbase
+#include <filesystem>
+#include <fstream>
+
+namespace Utils
+{
+
+/*
+ * Load the test account credentials from a location adjacent to the binary
+ * Currently only provides 2 accounts, will need to be extended for tests that require more than 2 agents.
+ */
+TestAccountCredentials LoadTestAccountCredentials()
+{
+	if (!std::filesystem::exists("test_account_creds.txt"))
+	{
+		constexpr const char* msg
+			= "test_account_creds.txt not found! This file must exist and must contain the following information:\n<DefaultLoginEmail> "
+			  "<DefaultLoginPassword>\n<AlternativeLoginEmail> <AlternativeLoginPassword>\n<SuperUserLoginEmail> <SuperUserLoginPassword>";
+		throw ExceptionWithCode(MultiplayerTestRunner::ErrorCodes::COULD_NOT_FIND_CREDENTIALS_FILE, msg);
+	}
+
+	std::ifstream CredsFile;
+	CredsFile.open("test_account_creds.txt");
+
+	std::string DefaultLoginEmail, DefaultLoginPassword, AlternativeLoginEmail, AlternativeLoginPassword, SuperUserLoginEmail, SuperUserLoginPassword;
+
+	CredsFile >> DefaultLoginEmail >> DefaultLoginPassword;
+	CredsFile >> AlternativeLoginEmail >> AlternativeLoginPassword;
+	CredsFile >> SuperUserLoginEmail >> SuperUserLoginPassword;
+
+	if (DefaultLoginEmail.empty() || DefaultLoginPassword.empty() || AlternativeLoginEmail.empty() || AlternativeLoginPassword.empty()
+		|| SuperUserLoginEmail.empty() || SuperUserLoginPassword.empty())
+	{
+		constexpr const char* msg
+			= "test_account_creds.txt must be in the following format:\n<DefaultLoginEmail> <DefaultLoginPassword>\n<AlternativeLoginEmail> "
+			  "<AlternativeLoginPassword>\n<SuperUserLoginEmail> <SuperUserLoginPassword>";
+		throw ExceptionWithCode(MultiplayerTestRunner::ErrorCodes::MALFORMED_CREDENTIALS_FILE, msg);
+	}
+
+	return TestAccountCredentials {DefaultLoginEmail.c_str(),
+								   DefaultLoginPassword.c_str(),
+								   AlternativeLoginEmail.c_str(),
+								   AlternativeLoginPassword.c_str(),
+								   SuperUserLoginEmail.c_str(),
+								   SuperUserLoginPassword.c_str()};
+}
+
+bool RequestPredicate(const csp::systems::ResultBase& Result)
```

Comment:
[Consider] Is this used anywhere anymore since the migration to futures? You might be able to delete it.

Reply:
https://github.com/magnopus-opensource/connected-spaces-platform/pull/558/commits/b9fd2380203022080eb0e51d1a5d29cd78874eae